src/rdb.c

```diff
@@ -3482,6 +3482,7 @@ static void backgroundSaveDoneHandlerSocket(int exitcode, int bysignal) {
     if (!bysignal && exitcode == 0) {
         serverLog(LL_NOTICE,
             "Background RDB transfer terminated with success");
+        server.lastsave = time(NULL);
```

Comment:
It may not be used here. backgroundSaveDoneHandlerSocket is not strictly an RDB file save.

Reply:
@enjoy-binbin you may need to clarify what you mean, from my understanding, backgroundSaveDoneHandlerSocket is handling saves to a socket while backgroundSaveDoneHandlerDisk is handling rdb file saves (if by file we mean to disk). I noticed that server.lastsave is updated in [backgroundSaveDoneHandlerDisk](https://github.com/redis/redis/blob/91863dd854feba7f75ae58976a920acb192a5b67/src/rdb.c#L2782) but not in the socket equivalent, I'm not sure why. Please let me know if I have misunderstood

Reply:
yean, you understanding is right. My point is, backgroundSaveDoneHandlerDisk is actually write a RDB file on the disk, and backgroundSaveDoneHandlerSocket was write the data to the socket and it is diskless. I'm not quite sure if it's the right thing to update it

Reply:
Yeah, I think the motivation is missing, @bentotten can you update the top comment for why we want to make this change?